rdm/render.py

```diff
@@ -34,30 +34,23 @@ def join_to(foreign_keys, table, primary_key='id'):
     return joined
 
 
-def md_indent(snippet, header_shift=0):
+def md_indent(snippet, header_shift = 0):
     lines = snippet.split('\n')
-    i = 0
     in_code = False
-    if header_shift < 0:
-        while i > header_shift:
-            i -= 1
-            for j in range(0, len(lines)):
-                if lines[j][0:3] == "```":
-                    in_code = not in_code
-                if in_code:
-                    continue
-                if lines[j][0] == '#':
-                    lines[j] = lines[j][1:].lstrip()
-    else:
-        while i < header_shift:
-            i += 1
-            for j in range(0, len(lines)):
-                if lines[j][0:3] == "```":
-                    in_code = not in_code
-                if in_code:
-                    continue
-                if lines[j][0] == '#':
-                    lines[j] = '#' + lines[j]
+    for j, line in enumerate(lines):
+        if line[0:3] == "```":
+            in_code = not in_code
+        if in_code:
+            continue
+        if header_shift < 0:
+            if line[0] == '#':
+                if line[0:abs(header_shift)+1] == '#' * (abs(header_shift) + 1):
+                    lines[j] = line[abs(header_shift):]
+                else:
+                    raise UserWarning('This will remove the heading indent entirely for one or more lines in your snippet')
```

Comment:
Interesting! I don't actually know too much about Python's warning system, but it seems like `UserWarning` is part of it. What lead you to `UserWarning` here? (It may be a valid choice, I'm just not sure. My initial thought would have been `ValueError` (https://docs.python.org/3/library/exceptions.html#ValueError))

Reply:
I had browsed through the list of exemption types and UserWarning seemed like the best option since it indicates a problem with something done by the user. Although I did not see ValueError which might actually be more appropriate for this case